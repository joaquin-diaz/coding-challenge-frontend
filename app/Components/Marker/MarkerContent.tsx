import * as React from 'react';
import styled from '@emotion/styled';

interface ContainerProps {
  hidden: boolean;
}

const Container = styled.div<ContainerProps>`
  position: absolute;
  background: white;
  box-shadow: 0 1px 6px 0 rgba(32, 33, 36, 0.28);
  box-sizing: border-box;

  bottom: 35px;
  width: 200px;
  left: -100px;
  right: 0;
  padding: 10px 20px;
  border-radius: 5px;
  font-size: 15px;

  ${props => (props.hidden ? 'opacity: 0;' : 'opacity: 1;')}
`;

interface MarkerContentProp {
  title: string;
  address: string;
  hidden: boolean;
}

const MarkerContent: React.FC<MarkerContentProp> = props => {
  return (
    <Container hidden={props.hidden}>
      {props.title} <br /> {props.address}
    </Container>
  );
};

export default MarkerContent;
