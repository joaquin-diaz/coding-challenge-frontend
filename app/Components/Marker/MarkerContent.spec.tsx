import * as React from 'react';

import MarkerContent from './MarkerContent';
import { mount } from 'enzyme';

describe('<MarkerContent />', () => {
  it('should render', () => {
    const wrapper = mount(<MarkerContent title="Something" address="Testy" hidden={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render hidden', () => {
    const wrapper = mount(<MarkerContent title="Something" address="Testy" hidden />);
    expect(wrapper).toMatchSnapshot();
  });
});
