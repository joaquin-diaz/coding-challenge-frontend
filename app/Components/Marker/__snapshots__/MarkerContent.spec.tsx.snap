// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MarkerContent /> should render 1`] = `
.emotion-0 {
  background: white;
  box-shadow: 0 1px 6px 0 rgba(32,33,36,0.28);
  box-sizing: border-box;
  border-radius: 5px;
  position: absolute;
  bottom: 35px;
  width: 200px;
  left: -100px;
  right: 0;
  padding: 10px 20px;
  font-size: 15px;
  z-index: 10;
  opacity: 1;
}

<MarkerContent
  address="Testy"
  hidden={false}
  title="Something"
>
  <Styled(div)
    hidden={false}
  >
    <div
      className="emotion-0"
      hidden={false}
    >
      Something
       
      <br />
       
      Testy
    </div>
  </Styled(div)>
</MarkerContent>
`;

exports[`<MarkerContent /> should render hidden 1`] = `
.emotion-0 {
  background: white;
  box-shadow: 0 1px 6px 0 rgba(32,33,36,0.28);
  box-sizing: border-box;
  border-radius: 5px;
  position: absolute;
  bottom: 35px;
  width: 200px;
  left: -100px;
  right: 0;
  padding: 10px 20px;
  font-size: 15px;
  z-index: 10;
  opacity: 0;
}

<MarkerContent
  address="Testy"
  hidden={true}
  title="Something"
>
  <Styled(div)
    hidden={true}
  >
    <div
      className="emotion-0"
      hidden={true}
    >
      Something
       
      <br />
       
      Testy
    </div>
  </Styled(div)>
</MarkerContent>
`;
